app/views/shared/_tag_manager.html.erb

```diff
@@ -1,4 +1,11 @@
 <% if ENV.include?("GOOGLE_TAG_MANAGER_ID") %>
+
+<script>
+  window.dataLayer = window.dataLayer || [
+    { "scrapbookId": "<%= current_scrapbook.id %>" }
+  ]
+</script>
```

Comment:
There's no way to extract it from the URL using GTM with the current whitelist. This makes it available in the GTM dataLayer, so that it can then be sent through to GA as a UserID (https://support.google.com/analytics/answer/3123662?hl=en)